src/net.h

```diff
@@ -695,28 +693,50 @@ class CNode
     std::vector<CAddress> vAddrToSend;
     CRollingBloomFilter addrKnown;
     bool fGetAddr{false};
-    std::set<uint256> setKnown;
     int64_t nNextAddrSend GUARDED_BY(cs_sendProcessing){0};
     int64_t nNextLocalAddrSend GUARDED_BY(cs_sendProcessing){0};
 
-    // inventory based relay
-    CRollingBloomFilter filterInventoryKnown GUARDED_BY(cs_inventory);
-    // Set of transaction ids we still have to announce.
-    // They are sorted by the mempool before relay, so the order is not important.
-    std::set<uint256> setInventoryTxToSend;
+    const bool m_addr_relay_peer;
+    bool IsAddrRelayPeer() const { return m_addr_relay_peer; }
+
     // List of block ids we still have announce.
     // There is no final sorting before sending, as they are always sent immediately
     // and in the order requested.
     std::vector<uint256> vInventoryBlockToSend GUARDED_BY(cs_inventory);
     CCriticalSection cs_inventory;
-    int64_t nNextInvSend{0};
+
+    struct TxRelay {
```

Comment:
Can you move this into CNodeState? They never belonged in CNode and the only reason they're still there is because its not been worth doing a free-standing PR to do it, but since you're moving them, might as well merge it all with TxDownloadState anyway.

Reply:
That turns out to be more invasive than I realized when we discussed offline -- fRelayTxes and the bloom filter are used in net.cpp, particularly for the inbound peer eviction logic, so refactoring to allow that to live in net_processing would be more complicated than I think makes sense for this PR.

Reply:
Done in #21160